Basic documentation that's missing from the website

bugarela opened the pull request:
Hello :octocat: 

This is a website revamp :sparkles:. I reorganized the whole structure, and added several new pages:
  - Introduction > Why?
  - Introduction > Quint Basics
  - Introduction > Language Basics
  - Using specifications > Checking properties
  - Using specifications > Model-based Testing
  - How the tools work > Model checkers
  - How the tools work > Understanding the simulator

You can check a preview here: https://gabriela-basic-docs--quint-docs.netlify.app/docs/getting-started

I'm looking for two things here:
1. **High level review**, making sure there is nothing too out of place. Considering that the current website has zero information about some of the stuff I'm adding, I'm looking for merge this soon and not go into too much nitpicking while the current state is lacking so much.
2. **Any sort of feedback** to be implemented on the future.

(1) is higher priority than (2) at this point, and we can always iterate after merging.

josef-widder reviewed (commented):
Looks great. I left some comments. For REPL, I guess it is untouched in this PR, so I left a comment on some other file. Sorry for that.

bugarela commented on docs/pages/docs/quint-basics.mdx:
```diff
@@ -0,0 +1,182 @@
+# Quint basics
```
Right! I actually got the same feedback from my partner, and thought about changing Quint Basics to "Basic Concepts". Maybe "What Quint does" is a good one also.

bugarela commented:
@josef-widder I've addressed the comments and ended up also moving "Language Basics" to the "Writing Specifications" section like this:
![image](https://github.com/user-attachments/assets/fd7eebe1-168e-4acb-822f-feff03f0151c)

Ivan didn't had time to review before his leave, but I think we are good to merge. Can you take another look and give me an approval if you are ok with proceding?
